src/components/Cards.tsx

```diff
@@ -1,129 +1,140 @@
-import styles from './CssModule/NewCard.module.scss';
-import { Trash } from './icons/Trash';
-import { Pencil } from './icons/Pencil';
-import React, { useState } from 'react';
+import styles from "./CssModule/NewCard.module.scss";
+import { Trash } from "./icons/Trash";
+import { Pencil } from "./icons/Pencil";
+import React, { useState } from "react";
 
 interface Props {
-	heading: string;
-	answer: string;
+  heading: string;
+  answer: string;
 }
 
 export const Cards = ({ heading, answer }: Props) => {
-	const [isEditCard, setIsEditCard] = useState(false);
-	const [inputValue, setInputValue] = useState(heading);
-	const [answerInputValue, setAnswerInputValue] = useState(answer);
-	const [prevInputValue, setPrevInputValue] = useState('');
-	const [prevAnswerInputValue, setAnswerPrevInputValue] = useState('');
-	const [flipCard, setFlipCard] = useState(false);
-	const [showError, setShowError] = useState(false);
+  const [isEditCard, setIsEditCard] = useState(false);
+  const [inputValue, setInputValue] = useState(heading);
+  const [answerInputValue, setAnswerInputValue] = useState(answer);
+  const [prevInputValue, setPrevInputValue] = useState("");
```

Comment:
**nitpick:** You don't have to keep previous values as it is available through props. I would keep only one local state with editable value as it will reduce complexity of your component

Reply:
I mean that you already have `heading`, and `answer` props. Try to update them directly in the cards array that you have in `App.tsx`.

I mean this array:
```
  const [cards, setCards] = useState(initialCards);
```